### ISSUE TITLE
Fix S6966 FP: Mock.Raise from Moq should be ignored

### ISSUE THREAD

**Issue by a10r** [SonarSource/sonar-dotnet#9697]
### Description

In an async test method, S6966 is raised for Moq's Mock.Raise method, suggesting to use RaiseAsync instead.

RaiseAsync does not work for normal events and requires a Task-returning event signature instead: https://github.com/moq/moq/pull/1313.

I've seen that this rule has a list of exceptions for specific APIs, so maybe it would make sense to add this case to that list.

### Repro steps

Consider this:

```cs
public interface IReproInterface
{
	event EventHandler SomethingHappened;
}

public class Repro
{
	[Test]
	public async Task Test()
	{
		await Task.Yield();
		
		var mock = new Mock<IReproInterface>();
		
		mock.Raise(i => i.SomethingHappened += null, new EventArgs()); // "Await RaiseAsync instead"
	}
}
```

### Expected behavior

Should not suggest RaiseAsync.

### Actual behavior

Suggests to use RaiseAsync instead of Raise.

### Known workarounds

--

### Related information

* C#/VB.NET Plugins version: SonarLint 8.6.0.10679
* Visual Studio version: 17.12.0
* MSBuild / dotnet version: dotnet 9.0.100
* Operating System: Windows 11

**Comment by sebastien-marichal** [comment]
Hello @a10r,

I confirm this is a false positive.

We might need to do a bit more than just add it to the exclusion list.
I am thinking about checking if the event handler returns a task; however, I am not sure this would add value to the rule.

We should also add an exclusion list in RSPEC.

**Comment by aKzenT** [comment]
Please also allow user to configure manual exclusions. We have internal APIs that trigger false positives on this.

**Comment by pavel-mikula-sonarsource** [comment]
Test PAVEL-46

**Comment by pavel-mikula-sonarsource** [comment]
Internal ticket [NET-1535](https://sonarsource.atlassian.net/browse/NET-1535)

[NET-1535]: https://sonarsource.atlassian.net/browse/NET-1535?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
